### PR TITLE
Update workflow files

### PR DESCRIPTION
**Pull request by AliAkberAakash**
## :scroll: Description
   _#skip-changelog_ 
 - Add condition to run e2e tests only when `SENTRY_AUTH_TOKEN` is not present
 - Remove unused env variable from `flutter.yml`

## :bulb: Motivation and Context
Fix to issue #733 

## :green_heart: How did you test it?
Merged with my main branch and ran workflows. Some tests are failing but I don't think they are for my changes.

## :pencil: Checklist
- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
@AliAkberAakash yes, you are right, it's a breaking change on the Flutter beta channel, working on it.

File: .github/workflows/flutter.yml [hunk 1]
```diff
@@ -89,8 +89,6 @@ jobs:
             min_coverage: 90
 
         - name: Build ${{ matrix.target }}
-          env:
-            SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
```
Comment (ueman) [inline .github/workflows/flutter.yml:93]: Shouldn't this be kept? Otherwise, the E2E tests doesn't work any more even if `SENTRY_AUTH_TOKEN` is available.

Comment (AliAkberAakash) [reply .github/workflows/flutter.yml:93]: Should I put this again?

Comment (marandaneto) [reply .github/workflows/flutter.yml:93]: @ueman why do we need this? if `SENTRY_AUTH_TOKEN` is given in the e2e_dart.yml? I don't see it being used.

Comment (ueman) [reply .github/workflows/flutter.yml:93]: Oh sorry, never mind then.

**Review by marandaneto — APPROVED** [review]

**Comment by marandaneto** [comment]
Thanks @AliAkberAakash

**Comment by AliAkberAakash** [comment]
> Thanks @AliAkberAakash

Thanks for merging it, I had already merged this with my main so wasn't able to work on the other issue.

**Pull request merged**